app/src/main/java/nu/yona/app/api/manager/impl/NotificationManagerImpl.java

```diff
@@ -197,10 +197,13 @@ public void onError(Object errorMessage)
 
 	public YonaMessages processYonaMessages(YonaMessages resultYonaMessages)
 	{
-		List<YonaMessage> listMessages = resultYonaMessages.getEmbedded().getYonaMessages();
-		for (YonaMessage message : listMessages)
+		if (resultYonaMessages.getEmbedded() != null && resultYonaMessages.getEmbedded().getYonaMessages() != null)
```

Comment:
Instead of testing for null, test the number of messages, something like this:
```
if (resultYonaMessages.getPage().getNumber() == 0)
{
    return resultYonaMessages;
}
```